operator/builtin/transformer/recombine/recombine.go

```diff
@@ -147,9 +152,19 @@ func (r *RecombineOperator) flushLoop() {
 		select {
 		case <-r.ticker.C:
 			r.Lock()
-			if err := r.flushCombined(); err != nil {
-				r.Errorw("Failed flushing", "error", err)
+			timeNow := time.Now()
+			for source, entries := range r.batchMap {
+				lastEntryTs := entries[len(entries)-1].Timestamp
+				timeSinceLastEntry := timeNow.Sub(lastEntryTs)
+				if timeSinceLastEntry > r.forceFlushTimeout {
+					err := r.flushSource(source)
+					if err != nil {
+						r.Errorf("there was error flushing combined logs %s", err)
+					}
+				}
```

Comment:
Nit: This is quite deeply nested. Perhaps slightly more idiomatic:

```suggestion
			timeNow := time.Now()
			for source, entries := range r.batchMap {
				lastEntryTs := entries[len(entries)-1].Timestamp
				timeSinceLastEntry := timeNow.Sub(lastEntryTs)
				if timeSinceLastEntry < r.forceFlushTimeout {
					continue
				}
				if err := r.flushSource(source); err != nil {
					r.Errorf("there was error flushing combined logs %s", err)
				}
```